File: javascript/resistor-color-duo/resistor-color-duo.js
const Encoding = {
  "black": 0,
  "brown": 1,
  "red": 2,
  "orange": 3,
  "yellow": 4,
  "green": 5,
  "blue": 6,
  "violet": 7,
  "grey": 8,
  "white": 9
}

export const decodedValue = (array) => {

  /*Not needed because only two are tested and accepted
  function recursiveValueAdder(array){
    if(array.length <= 0) return ''
    console.log(Encoding[array[0]])
    return `${Encoding[array.shift()]}${recursiveValueAdder(array)}`
  }
  return parseInt(recursiveValueAdder(array))*/

  return parseInt(`${Encoding[array[0]]}${Encoding[array[1]]}`)

};

File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: true

class ResistorColorDuo
  @encoding = {
    'black': 0,
    'brown': 1,
    'red': 2,
    'orange': 3,
    'yellow': 4,
    'green': 5,
    'blue': 6,
    'violet': 7,
    'grey': 8,
    'white': 9
  }.freeze

  def self.value(_params)
    "#{@encoding[_params[0].to_sym]}#{@encoding[_params[1].to_sym]}".to_i
  end
end
